sgt-back-end/index.js

```diff
@@ -0,0 +1,242 @@
+const pg = require('pg');
+const express = require('express');
+
+const PORT = 3000;
+const app = express();
+const JSONparser = express.json();
+app.use(JSONparser);
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+
+  const errMessage = inputChecks(req, res, ['id']);
+  if (errMessage) {
+    res.status(400).json({
+      error: errMessage
+    });
+    return;
+  }
+
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+
+  const params = [gradeId];
+  // https://node-postgres.com/features/queries#parameterized-query
+  db.query(sql, params)
+    .then(result => {
+      // https://node-postgres.com/api/result
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with gradeId = ${gradeId}`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.get('/api/grades/', (req, res, next) => {
+  // checks to break if unnecessary params perhaps?
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result;
+      res.status(200).json(grade.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
```

Comment:
✔

---

```js
// checks to break if unnecessary params perhaps?
```

It's generally not necessary to check for parameters outside of what the route needs to complete its job.